src/main/java/org/commcare/formplayer/session/MenuSession.java

```diff
@@ -153,8 +153,8 @@ public boolean handleInput(String input, boolean needsDetail, boolean confirmed,
             boolean addBreadcrumb = true;
             if (screen instanceof EntityScreen) {
                 EntityScreen entityScreen = (EntityScreen)screen;
-                boolean autoLaunch = entityScreen.getAutoLaunchAction() != null;
-                addBreadcrumb = !(autoLaunch && allowAutoLaunch);
+                boolean autoLaunch = entityScreen.getAutoLaunchAction() != null && allowAutoLaunch;
+                addBreadcrumb = !autoLaunch;
                 if (input.startsWith("action ") || (autoLaunch) || !confirmed) {
```

Comment:
Today the only place where `allowAutoLaunch` is false , `confirmed` is also set to `false` resulting in this line evaluating to `true` even when we don't factor in `allowAutoLaunch` in the `autoLaunch` flag.